release-notes/SRM/SRM_1.2.29.md

```diff
@@ -0,0 +1,118 @@
+
+In addition, in case invalid data is provided to the *EngineExtensionMethods.SynchronousAddService* method, it will now throw a *ServiceInvalidDataException* with details about the unexpected data.
+
+#### SRMSettableServiceState objects no longer used [ID_34788]
+
+To improve performance, the SRM Framework no longer creates and updates *SRMSettableServiceState* objects during booking execution. Previously, these objects were only used for logging purposes.
+
+#### Superfluous ValidateContributingResource method removed [ID_34826]
+
+The *ValidateContributingResource* method in the *AssignResourceRequest* class has been removed. This method was developed as a workaround for an issue that was fixed in DataMiner 10.1.8/10.2.0 and is consequently superfluous now.
+
+#### Extended Booking Failure script now used whenever LSO action fails [ID_34848]
+
+When an LSO action fails, the Extended Booking Failure script will now be called. Up to now, this was only called when a booking failed to start.
+
+The error data in the script will hold a new value of LsoFailed for these new cases.
+
+The example script *SRM_BookingStartFailureTemplate* has been updated accordingly.
+
+#### 'DMA IDs to Store Reservations' renamed to 'Hosting DMA IDs to Orchestrate Reservation' [ID_34882]
+
+In the Booking Manager app, the parameter *DMA IDs to Store Reservations* has been renamed to *Hosting DMA IDs to Orchestrate Reservation*.
+
+#### SrmServiceInfo object now has name of service instead of booking [ID_34951]
+
+When a booking is created, either silently or with the Booking Wizard, it is possible to define the name for a persistent service. From now on, the corresponding *SrmServiceInfo* object will also be created with that service name. Previously, the name of the booking was used instead.
+
+#### SLA improvements [ID_34976]
+
+To prevent possible issues and add several improvements, the SLA implementation in the SRM Framework has been reviewed.
+
+- Problems with an SLA element can now no longer cause orchestration to fail.
+- Logging has been improved.
+- A possible issue has been resolved where SRM incorrectly thought an SLA element had not been created.
+- SLA elements will now only be created when a booking starts.
+
+### Fixes
+
+#### Booking Manager: Rescue button displayed when not relevant [ID_34618]
+
+In the Booking Manager app, it could occur that the *Rescue* button was also displayed for bookings that were not in an interrupted state.
+
+#### AddResource not adding resources to service of active booking [ID_34628]
+
+When a resource was added via the *AddResource* method to a booking that had not started yet and to its service definition, it could occur that the relevant service was added to the service of the booking, but when the booking started, the other assigned resources were not added to the service. This only happened when the service was created in advance or for contributing bookings.
```

Comment:
Peter has reviewed the RN , because it was a bit confusing

Reply:
I've adjusted the RN. Could you take a look?